src/RADOS/RADOSImpl.cc

```diff
@@ -0,0 +1,117 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2004-2012 Sage Weil <sage@newdream.net>
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+
+#include <boost/system/system_error.hpp>
+
+#include "common/common_init.h"
+
+#include "global/global_init.h"
+
+#include "RADOSImpl.h"
+
+namespace RADOS {
+namespace _ {
+
+RADOS::RADOS(boost::asio::io_context& ioctx,
```

Comment:
Sweeping Generality(TM): There's a lot of code in this ctor. I don't see anything "wrong" with that, but sometimes it's nice to have a private member function that helps to handle complex setup away from other things that may be added later (logging, etc.).

Reply:
Fixed in other PR...